tools/llzk-tblgen/CommonCAPIGen.h

```diff
@@ -0,0 +1,661 @@
+} // namespace clang
+
+// Shared command-line options used by all CAPI generators
+extern llvm::cl::OptionCategory OpGenCat;
+extern llvm::cl::opt<std::string> DialectName;
+extern llvm::cl::opt<std::string> FunctionPrefix;
+
+// Shared flags for controlling code generation
+extern llvm::cl::opt<bool> GenIsA;
+extern llvm::cl::opt<bool> GenOpBuild;
+extern llvm::cl::opt<bool> GenOpOperandGetters;
+extern llvm::cl::opt<bool> GenOpOperandSetters;
+extern llvm::cl::opt<bool> GenOpAttributeGetters;
+extern llvm::cl::opt<bool> GenOpAttributeSetters;
+extern llvm::cl::opt<bool> GenOpRegionGetters;
+extern llvm::cl::opt<bool> GenOpResultGetters;
+extern llvm::cl::opt<bool> GenTypeOrAttrGet;
+extern llvm::cl::opt<bool> GenTypeOrAttrParamGetters;
+extern llvm::cl::opt<bool> GenExtraClassMethods;
+
+/// @brief Convert names separated by underscore or colon to PascalCase.
+/// @param str The input string to convert (may contain underscores or colons)
+/// @return The converted PascalCase string
+///
+/// Examples:
+///   "no_inline" -> "NoInline"
+///   "::llzk::boolean::Type" -> "LlzkBooleanType"
+inline std::string toPascalCase(mlir::StringRef str) {
+  if (str.empty()) {
+    return "";
+  }
+
+  std::string result;
+  result.reserve(str.size());
+  llvm::raw_string_ostream resultStream(result);
+  bool capitalizeNext = true;
+
+  for (char c : str) {
+    if (c == '_' || c == ':') {
+      capitalizeNext = true;
+    } else {
+      resultStream << (capitalizeNext ? llvm::toUpper(c) : c);
+      capitalizeNext = false;
+    }
+  }
+
+  return result;
+}
+
+/// @brief Check if a C++ type is a known primitive type
+/// @param cppType The C++ type string to check
+/// @return true if the type is a primitive (bool, void, int, etc.)
+///
+/// @note This function must be called on the CPP type because after converting to CAPI type, some
+/// things like APInt become primitive which can lead to missing wrap/unwrap functions.
+inline bool isPrimitiveType(mlir::StringRef cppType) {
+  cppType.consume_front("::");
+  return llvm::StringSwitch<bool>(cppType)
+      .Case("bool", true)
+      .Case("void", true)
+      .Case("int", true)
+      .Case("unsigned", true)
+      .Case("size_t", true)
+      .Case("intptr_t", true)
+      .Case("int32_t", true)
+      .Case("int64_t", true)
+      .Case("uint32_t", true)
+      .Case("uint64_t", true)
+      .Default(false);
+}
+
+/// @brief Check if a token text represents a C++ modifier/specifier keyword
+/// @param tokenText The token to check
+/// @return true if the token is a C++ modifier (inline, static, virtual, etc.)
+inline bool isCppModifierKeyword(mlir::StringRef tokenText) {
+  return llvm::StringSwitch<bool>(tokenText)
+      .Case("inline", true)
+      .Case("static", true)
+      .Case("virtual", true)
+      .Case("explicit", true)
+      .Case("constexpr", true)
+      .Case("consteval", true)
+      .Case("extern", true)
+      .Case("mutable", true)
+      .Case("friend", true)
```

Comment:
Does this only need to specify leading modifiers and not trailing ones (e.g., `const`)?